src/pages/Products/ProductPage.js

```diff
@@ -2,32 +2,54 @@ import "../Products/ProductPage.css";
 import { Filter, ProductCard } from "../../components";
 import axios from "axios";
 import { useEffect, useState } from "react";
+import { useProduct } from "../../contexts";
+import { getfilteredCategoryProducts,getfilteredPriceProducts,getfilteredRatingProducts,getSortedProducts } from "../../utilities/filterFunctions";
+import { products } from "../../backend/db/products";
 
 export function ProductPage() {
-    const [products, setProducts] = useState([]);
+    const {state,dispatch} = useProduct();
 
-    useEffect(() => {
-        (async () => {
-            try {
-                const productsData = await axios.get("/api/products");
+    const filteredPriceProducts=getfilteredPriceProducts(state.products,state.price);
+    console.log(state.products,"heloooo")
+    
+    const filteredCategoryProducts=getfilteredCategoryProducts(filteredPriceProducts,
```

Comment:
destructure state.category.
Suggestion: name your state according to the context it is refering to for example for filter-context the state may be named as filterState instead of just state. this will help avoid confusions when multiple contexts are present.